package com.cart_service.controller;

import com.cart_service.dto.AddtoCartRequest;
import com.cart_service.entity.Cart;
import com.cart_service.service.CartService;
import org.springframework.http.HttpHeaders;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@CrossOrigin(exposedHeaders = "X-CART-ID")
@RestController
@RequestMapping("/api/v1/cart")
public class CartController {

    private final CartService cartService;

    public CartController(CartService cartService) {
        this.cartService = cartService;
    }

    @PostMapping("/add")
    public ResponseEntity<String> addToCart(
            @RequestHeader(value = "X-CART-ID", required = false) String uuid,
            @RequestBody AddtoCartRequest request
    ) {
        System.out.println("CART CONTROLLER REACHED");
        Cart cart = cartService.addToCart(uuid, request);

        HttpHeaders headers = new HttpHeaders();
        headers.set("X-CART-ID", cart.getUuid());

        return ResponseEntity.ok()
                .headers(headers)
                .body("product added successfully");
    }


    @GetMapping("/{uuid}")
    public Cart getCart(@PathVariable String uuid) {
        return cartService.getCart(uuid);

    }

    @DeleteMapping("/{uuid}")
    public void clearCart(@PathVariable String uuid) {
        cartService.clearCart(uuid);
    }
}
